Wikipedia/Code/MWKHistoryList.m

```diff
@@ -154,6 +154,11 @@ - (void)addPageToHistoryWithURL:(NSURL *)URL {
     NSDate *now = [NSDate date];
 
     WMFArticle *article = [self.dataStore fetchOrCreateArticleWithURL:URL];
+    
+    if (!article.displayTitle) {
```

Comment:
This check will skip history for any article opened via a universal link from Safari that hasn't been viewed in the app yet. The data store will only have a `WMFArticle` object with a display title if the article appeared in the feed or search results.

As a result, with this change, if you do a clean install of the app, exit the app, enter an article URL into Safari that isn't in the feed (I tested with https://en.wikipedia.org/wiki/Barack_Obama), then tap "Open" on the banner on the top of the page, it won't show up in history.